src/utils.jl

```diff
@@ -84,16 +85,27 @@ function build_layer_from_vec!(
   return sizecuArray
 end
 
+function build_layer_from_vec!(
+  cuArray::Array{T, N} where {N},
```

Comment:
is this `where` at the right place?

Reply:
Also, don't call it cuArray if it's not necessarily a CuArray.

Reply:
I used `(.., array::A) where A<:AbstractArray{T,N}`